framework/log/Dispatcher.php

```diff
@@ -80,7 +80,7 @@ public function __construct($config = [])
     {
         // ensure logger gets set before any other config option
         if (isset($config['logger'])) {
-            $this->setLogger($config['logger']);
+            $this->setLogger(Yii::createObject($config['logger']));
```

Comment:
I updated the tests to use the config semantics before I saw this comment. Would you prefer I back those changes out and add the is_array check?